subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/logging/GroupedOutputFixture.java

```diff
@@ -26,65 +32,52 @@
+    private final static String BUILD_STATUS_FOOTER = "BUILD SUCCESSFUL";
+    private final static String BUILD_FAILED_FOOTER = "FAILURE:";
 
     /**
      * Various patterns to detect the end of the task output
      */
     private final static String END_OF_TASK_OUTPUT = TASK_HEADER + "|" + BUILD_STATUS_FOOTER + "|" + BUILD_FAILED_FOOTER;
 
-    /**
-     * Patterns to remove from the output: cursor movement, progress bar, work-in-progress items and scrolling the console
-     */
-    private final static String CURSOR_BACK_PATTERN = "\u001b\\[\\d+D";
-    private final static String CURSOR_UP_PATTERN = "\u001b\\[\\d+A";
-    private final static String CURSOR_DOWN_PATTERN = "\u001b\\[\\d+B";
-    private final static String CURSOR_MOVEMENT_PATTERN = CURSOR_BACK_PATTERN + "|" + CURSOR_DOWN_PATTERN + "|" + CURSOR_UP_PATTERN;
-
-    private final static String PROGRESS_BAR_PATTERN = "\u001b\\[1m<[-=]*> \\d+% (INITIALIZING|CONFIGURING|EXECUTING) \\[\\d+s\\]\u001b\\[m";
+    private final static String PROGRESS_BAR_PATTERN = "<[-=]*> \\d+% (INITIALIZ|CONFIGUR|EXECUT)ING \\[((\\d+h )? \\d+m )?\\d+s\\]";
     private final static String WORK_IN_PROGRESS_PATTERN = "\u001b\\[\\d+m> (IDLE|[:a-z][\\w\\s\\d:]+)\u001b\\[\\d*m";
-
     private final static String SCROLLING_WORK_IN_PROGRESS_PATTERN = "(\u001b\\[0K\\n)+\u001b\\[\\d+A";
 
+    private final static String WORK_IN_PROGRESS_AREA_PATTERN = PROGRESS_BAR_PATTERN + "|" + WORK_IN_PROGRESS_PATTERN + "|" + SCROLLING_WORK_IN_PROGRESS_PATTERN;
+
     /**
      * Pattern to extract task output.
      */
     private static final Pattern TASK_OUTPUT_PATTERN;
-    private static String precompiledPattern;
 
     static {
-        precompiledPattern = "(?ms)";
+        String precompiledPattern = "(?ms)";
         precompiledPattern += TASK_HEADER;
         // Capture all output, lazily up until two new lines and an END_OF_TASK designation
         precompiledPattern += "((.|\\n)*?(?=[^\\n]*?" + END_OF_TASK_OUTPUT + "))";
         TASK_OUTPUT_PATTERN = Pattern.compile(precompiledPattern);
     }
 
-    /**
-     * Don't need this if we parse all of the output during construction
-     */
-    private final String output;
+
+    private final String originalOutput;
     private Map<String, GroupedTaskFixture> tasks;
 
     public GroupedOutputFixture(String output) {
-        this.output = output;
+        this.originalOutput = output;
         parse(output);
     }
 
     private void parse(String output) {
-        tasks = new HashedMap();
-        String g = output.replaceAll(SCROLLING_WORK_IN_PROGRESS_PATTERN, "").replace(ERASE_TO_END_OF_LINE, "").replaceAll(CURSOR_MOVEMENT_PATTERN, "").replaceAll(PROGRESS_BAR_PATTERN, "").replaceAll(WORK_IN_PROGRESS_PATTERN, "");
-        Matcher matcher = TASK_OUTPUT_PATTERN.matcher(g);
+        tasks = new HashMap<String, GroupedTaskFixture>();
+
+        String stripedOutput = stripAnsiCodes(stripWorkInProgressArea(output));
```

Comment:
Unfortunately, we have to strip the work in progress items before removing the ANSI code as it will remove unwanted string (test output is `ClassName > test FAILED`, the string `> test` is removed is we don't account for the ANSI code).